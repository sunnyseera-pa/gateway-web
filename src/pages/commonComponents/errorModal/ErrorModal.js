import PropTypes from 'prop-types';
import React from 'react';
import { Button } from 'react-bootstrap';
import AlertModal from '../AlertModal';

const cmsURL = require('../../commonComponents/BaseURL').getCMSURL();

const ErrorModal = ({ onClose, ...outerProps }) => {
	const [show, setShow] = React.useState(true);

	const handleClose = React.useCallback(() => {
		if (onClose) onClose();
		setShow(false);

		//For analytics dashboard (at least), history push doesn't work

		// history.push({ pathname: '/' });
		window.location.pathname = '/';
	}, []);

	return (
		<AlertModal
			header='Oops! Something went wrong!'
			body={
				<>
					This issue has been automatically reported to our team!
					<br />
					If this issue continues, please contact support by clicking <a href={`${cmsURL}/HDRUKGatewaySupportPortal`}>here.</a>
				</>
			}
			footer={
				<Button onClick={handleClose} data-testid='close-button'>
					Close
				</Button>
			}
			variant='error'
			onHide={handleClose}
			show={show}
			{...outerProps}
		/>
	);
};

ErrorModal.propTypes = {
	onClose: PropTypes.func,
};

export default ErrorModal;
