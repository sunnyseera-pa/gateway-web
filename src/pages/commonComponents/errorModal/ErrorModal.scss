.modal-footer {
	border-top: none !important;
	padding-left: 32px !important;
	padding-right: 32px !important;
}
