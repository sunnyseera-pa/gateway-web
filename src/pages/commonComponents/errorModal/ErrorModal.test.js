import { render } from '@testing-library/react';
import React from 'react';
import ErrorModal from '.';

const onClose = jest.fn();

const props = {
    onClose,
};

let wrapper;
let containerDiv;
let closeButton;

describe('Given the ErrorModal component', () => {
    describe('When it is rendered', () => {
        beforeAll(() => {
            containerDiv = createPortalContainer();

            wrapper = render(<ErrorModal {...props} container={containerDiv} />);
        });

        afterAll(() => {
            removePortalContainer(containerDiv);
        });

        it('Then matches the previous snapshot', () => {
            expect(containerDiv).toMatchSnapshot();
        });

        it('Then has the correct header', () => {
            expect(wrapper.getByTestId('modal-header').textContent).toBe('Oops! Something went wrong!');
        });

        it('Then has the correct body', () => {
            expect(wrapper.getByTestId('modal-body').textContent).toBe(
                'This issue has been automatically reported to our team!If this issue continues, please contact support by clicking here.'
            );
        });

        it('Then has the correct footer', () => {
            expect(wrapper.getByTestId('modal-footer').textContent).toBe('Close');
        });

        describe('And the modal is closed', () => {
            beforeEach(() => {
                closeButton = wrapper.getByTestId('close-button');

                fireEvent.click(closeButton);
            });

            afterAll(() => {
                onClose.mockReset();
            });

            it('Then hides the modal', () => {
                expect(wrapper.container.querySelector('.show')).toBeNull();
            });

            it('Then calls onClose', () => {
                expect(onClose).toHaveBeenCalled();
            });

            it('Then hides the modal', () => {
                expect(location.pathname).toEqual('/');
            });
        });
    });

    describe('And the modal is closed without onClose', () => {
        beforeAll(() => {
            containerDiv = createPortalContainer();

            wrapper = render(<ErrorModal container={containerDiv} />);

            closeButton = wrapper.getByTestId('close-button');
            fireEvent.click(closeButton);
        });

        afterAll(() => {
            removePortalContainer(containerDiv);
        });

        it('Then doesnt call onClose', () => {
            expect(onClose).toHaveBeenCalledTimes(0);
        });
    });
});
