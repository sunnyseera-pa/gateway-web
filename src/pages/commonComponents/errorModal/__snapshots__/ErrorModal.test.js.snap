// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the ErrorModal component When it is rendered Then matches the previous snapshot 1`] = `
<div
  class="modal-open"
  style=""
>
  <div
    class="fade modal-backdrop show"
  />
  <div
    aria-labelledby="contained-modal-title-vcenter"
    aria-modal="true"
    class="fade modal show"
    role="dialog"
    style="display: block;"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-dialog-centered"
      variant="error"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title h4"
            data-testid="modal-header"
            id="contained-modal-title-vcenter"
          >
            Oops! Something went wrong!
          </div>
        </div>
        <div
          class="gray800-14 css-1axlqcz-AlertModal modal-body"
          data-testid="modal-body"
        >
          This issue has been automatically reported to our team!
          <br />
          If this issue continues, please contact support by clicking 
          <a
            href="http://localhost:3001/HDRUKGatewaySupportPortal"
          >
            here.
          </a>
        </div>
        <div
          class="css-n34xqh-AlertModal modal-footer"
          data-testid="modal-footer"
        >
          <button
            class="btn btn-primary"
            data-testid="close-button"
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
